utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/encode/BestEffortJsonSchemaEncoder.scala

```diff
@@ -105,9 +95,18 @@ class BestEffortJsonSchemaEncoder(validationMode: ValidationMode) {
   private def error(str: String): Invalid[NonEmptyList[String]] = Invalid(NonEmptyList.of(str))
 
   private def error(runtimeType: String, schema: String, fieldName: Option[String]): Invalid[NonEmptyList[String]] =
-    Invalid(NonEmptyList.of(s"Not expected type: ${runtimeType} for field${fieldName.map(f => s": '$f'").getOrElse("")} with schema: $schema"))
+    Invalid(NonEmptyList.of(s"Not expected type: $runtimeType for field${fieldName.map(f => s": '$f'").getOrElse("")} with schema: $schema."))
+
+  private[encode] def encodeWithJsonValidation(value: Any, schema: Schema, fieldName: Option[String] = None): EncodeOutput =
+    encode(value, schema, fieldName).andThen { result =>
+      val json = JsonSchemaUtils.circeToJson(result)
+      schema
+        .validateData(json)
+        .leftMap(errorMsg => NonEmptyList.of(errorMsg))
+        .map(_ => result) //we return base json (e.g. without felt defaults - there is no need to put these data as output)
+    }
 
-  def encode(value: Any, schema: Schema, fieldName: Option[String] = None): EncodeOutput = {
+  private[encode] def encode(value: Any, schema: Schema, fieldName: Option[String] = None): EncodeOutput = {
```

Comment:
Can we make this method private, and change unit tests to invoke this one `private[encode] def encodeWithJsonValidation` instead?

Reply:
Hmmm, Yes I can ;)